moveit_kinematics/cached_ik_kinematics_plugin/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+find_package(Boost COMPONENTS filesystem program_options REQUIRED)
+
+set(MOVEIT_LIB_NAME moveit_cached_ik_kinematics_base)
+add_library(${MOVEIT_LIB_NAME} src/ik_cache.cpp)
+set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
+target_link_libraries(${MOVEIT_LIB_NAME}
+    ${Boost_FILESYSTEM_LIBRARY}
+    ${catkin_LIBRARIES})
+install(TARGETS ${MOVEIT_LIB_NAME} LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+
+
+set(MOVEIT_LIB_NAME moveit_cached_ik_kinematics_plugin)
+add_library(${MOVEIT_LIB_NAME} src/cached_ik_kinematics_plugin.cpp)
+set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION})
+target_link_libraries(${MOVEIT_LIB_NAME}
+    moveit_cached_ik_kinematics_base
+    moveit_kdl_kinematics_plugin
+    moveit_srv_kinematics_plugin
+    ${OMPL_LIBRARIES}
+    ${catkin_LIBRARIES})
+if(trac_ik_kinematics_plugin_FOUND)
+    target_link_libraries(${MOVEIT_LIB_NAME} trac_ik_kinematics_plugin)
+    set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES COMPILE_DEFINITIONS "CACHED_IK_KINEMATICS_TRAC_IK")
+endif()
+install(TARGETS ${MOVEIT_LIB_NAME} LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+
+# This is just for testing purposes; the arms from Universal Robots have
+# analytic solvers, so caching just adds extra overhead.
+find_package(ur_kinematics QUIET)
```

Comment:
all UR references belong in [moveit_kinematics_tests](https://github.com/ros-planning/moveit_kinematics_tests/) so that we don't require that robot (and its dependencies) must be installed to use moveit. do you mind making a folder in that repo called ``cached_ik_kinematics_test``?

Reply:
The UR code is an optional dependency, not a requirement. I can move the code to  moveit_kinematics_tests, but the repo description says it's for unit tests. The UR code isn't a unit test.